server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -228,6 +228,9 @@
         patch :update, on: :member
       end
 
+      get 'stages/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/instance' => 'stages#show', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, pipeline_counter: PIPELINE_COUNTER_FORMAT, stage_name: STAGE_NAME_FORMAT, stage_counter: STAGE_COUNTER_FORMAT}, as: :stage_instance_by_counter_api
+      get 'stages/:pipeline_name/:stage_name/history' => 'stages#history', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, stage_name: STAGE_NAME_FORMAT}, as: :stage_history_api
+
```

Comment:
Fair point @arvindsv, I'm curious why the counters are defined as `PIPELINE_COUNTER_FORMAT = STAGE_COUNTER_FORMAT = /-?\d+/` When it can just be `/[0-9]+/`

Is this overloaded for some other purposes? I'd like to simplify the counter constraints if at all possible.